dlrs/main/pages/managelookuprollupsummaries_New.page

```diff
@@ -57,21 +57,41 @@
                 </apex:pageBlockButtons>
 
                 <apex:pageBlockSection title="Information" columns="2">
-                    <apex:pageBlockSectionItem>
-                        <apex:outputLabel value="Lookup Rollup Summary Name" />
+                    <apex:pageBlockSectionItem id="labelInputSection">
+                        <apex:outputLabel value="Lookup Rollup Summary Label" />
                         <apex:outputPanel>
                             <div class="requiredInput">
                                 <div class="requiredBlock" />
-                                <apex:inputText size="32" value="{!LookupRollupSummary.Label}" />
+                                <apex:inputText size="32" maxlength="40" value="{!LookupRollupSummary.Label}" id="labelInput"/>
                             </div>
                         </apex:outputPanel>
                     </apex:pageBlockSectionItem>
                     <apex:pageBlockSectionItem>
-                        <apex:outputLabel value="Lookup Rollup Summary Unique Name" />
+                        <apex:outputLabel value="Lookup Rollup Summary API Name" />
                         <apex:outputPanel>
                             <div class="requiredInput">
                                 <div class="requiredBlock" />
-                                <apex:inputText size="32" value="{!LookupRollupSummary.DeveloperName}" disabled="{!LookupRollupSummary.Id!=null}" />
+                                <script>
+                                    function suggestApiName(labelElementId, nameElementId) {
+                                        let labelElement = document.getElementById(labelElementId);
+                                        let nameElement = document.getElementById(nameElementId);
+
+                                        if (!nameElement.value) {
+                                            nameElement.value = convertLabelToApiName(labelElement.value);
+                                        }
+                                    }
+
+                                    function convertLabelToApiName(label) {
+                                        name = label.trim();
+                                        name = name.replaceAll(' ', '_');
+                                        name = name.replaceAll(/__+/g, '_');
+                                        name = [...name.matchAll(/[a-zA-Z0-9_]/g)].join('');
```

Comment:
While regex are helpful, they are hard to read/edit. Please consider variablenames for the patterns (e.g. `var OneOrMoreSpacesPattern = / +/g`) and add documentation to the convertLabel function  
By using `OneOrMoreSpacesPattern`, you can avoid using line 87.

line 88 is removing special chars, right? line 89 escapes me -.-

Let's try to be more expressive, please.